eole/models/model.py

```diff
@@ -868,12 +867,19 @@ def __init__(self, **kwargs):
             raise ValueError("DecoderModel should not be used" "with an encoder")
         if self.decoder is None:
             raise ValueError("DecoderModel requires a Decoder")
+        if self.add_estimator:
+            self.estimator = FeedForward(self.hidden_size)
 
     @classmethod
     def build_blocks(cls, model_config, vocabs, running_config=None):
         tgt_emb = build_tgt_emb(model_config, vocabs, running_config=running_config)
         decoder = build_decoder(model_config, running_config=running_config)
-        return cls(decoder=decoder, tgt_emb=tgt_emb)
+        return cls(
+            decoder=decoder,
+            tgt_emb=tgt_emb,
+            add_estimator=model_config.add_estimator,
```

Comment:
Feels weird to have decoder/tgt_emb initialized here, but estimator initialized in the class __init__. Maybe not this PR but we shall probably remove the `build_xxx` functions and properly factorize those within `Model` classes.

Reply:
I agree, but since it was the same way for the encoderdecoder model, let's keep this refactoring for another PR